Tests/NIOTests/EventLoopFutureTest.swift

```diff
@@ -82,6 +82,57 @@ class EventLoopFutureTest : XCTestCase {
             XCTFail("error of wrong type \(e)")
         }
     }
+    
+    func testReduceWithAllSuccesses() throws {
+        let eventLoop = EmbeddedEventLoop()
+        let promises: [EventLoopPromise<Int>] = (0..<5).map { (_: Int) in eventLoop.newPromise() }
+        let futures = promises.map { $0.futureResult }
+        
+        let fN: EventLoopFuture<[Int]> = EventLoopFuture<[Int]>.reduce([], futures, eventLoop: eventLoop, reducer: {$0 + [$1]})
+        _ = promises.map { $0.succeed(result: (1)) }
```

Comment:
would you mind making this succeed the promises with different values (maybe 1...5) so we can see that the associativity is correct?

Reply:
Sure